src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,960 @@
+        self.__assert_installation_truncated_msg_fields(installation_msg, installation_truncated_msg)
+
+    def test_only_installation_patches_large_size_limit_truncated(self):
+        """ Perform truncation on very large installation patches for time performance.
+        Input: xxx installation patches
+        Output:
+        operation: Installation,
+        installation substatus name: PatchInstallationSummary,
+        complete status file size > 128kb,
+        truncated status file size < 126kb < 128kb,
+        truncated status file size < completed status file size,
+        installation substatus status: warning,
+        installation errors code: 2 (warning),
+        installation errors details count: 0,
+        installation message json fields == truncated installation message json fields. """
+
+        self.runtime.execution_config.operation = Constants.INSTALLATION
+        self.runtime.status_handler.set_current_operation(Constants.INSTALLATION)
+
+        patch_count = 100000
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_install_status(test_packages, test_package_versions, Constants.INSTALLED)
+        self.runtime.status_handler.set_installation_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)
+
+        installation_msg = self.__get_message_json_from_substatus(complete_substatus_file_data)
+        self.__assert_patch_summary_from_status(complete_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_SUCCESS, patch_count, installation_msg, under_internal_size_limit=False)
+
+        # Assert installation truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            truncated_substatus_file_data = json.load(file_handle)
+
+        installation_truncated_msg = self.__get_message_json_from_substatus(truncated_substatus_file_data)
+        # Assert truncated status file size < completed status file size
+        self.assertTrue(len(json.dumps(truncated_substatus_file_data)) < len(json.dumps(complete_substatus_file_data)))
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_WARNING, patch_count, installation_truncated_msg)
+
+        # Assert all installation fields in the message json are equal in both status files
+        self.__assert_installation_truncated_msg_fields(installation_msg, installation_truncated_msg)
+
+    def test_only_installation_patches_over_size_limit_with_exceptions_truncated(self):
```

Comment:
this unit test is needed because we need to assert when substatus status is error, we do not change it to warning and i have combine this test with __assert_patch_summary_from_status and add assertions for truncation log.
done